### PR TITLE
Add sections to Python | Functions and edit article for clarity.

### PR DESCRIPTION
**Pull request by kaifee-haque**
### Description

- Added the following sections to [Python | Functions](https://www.codecademy.com/resources/docs/python/functions):
  - Keyword Arguments
  - Dynamically Passing Arguments
- Expanded on Functions with Varying Number of Arguments
- Edited some sections for clarity and fixed a mistake in Anonymous Functions

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@kaifee-haque, I'll go ahead and give your PR its first review. Expect something in the next 24 hours.

File: content/python/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/python/concepts/functions/functions.md:45]:
```suggestion
Functions often process data to achieve their intended task. Each value required by a function is passed as an parameter when that function is called.
```

Comment (SSwiniarski) [reply content/python/concepts/functions/functions.md:45]: The section talks about function parameters. Calling them arguments here is needlessly confusing.

File: content/python/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/python/concepts/functions/functions.md:49]:
```suggestion
For example, the function below contains parameters for a `character`, a `setting`, and a `special_skill`, which are used as inputs to write the first sentence of a book.
```

File: content/python/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/python/concepts/functions/functions.md:133]:
```suggestion
multiply(21, 24) # Output 504
```

File: content/python/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/python/concepts/functions/functions.md:134]:
```suggestion
multiply(10, 5, 3, 6, 17) # Output 15300
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@kaifee-haque, I made a few suggestions. Once they're resolved we can have a second review.

**Comment by sonnynomnom** [comment]
@kaifee-haque any updates?

**Comment by kaifee-haque** [comment]
Hi, sorry for the delay! Feedback has been implemented.

**Comment by Dusch4593** [comment]
@SSwiniarski do the changes look ok to clear for the next review?

**Review by SSwiniarski — APPROVED** [review]
@kaifee-haque , @Dusch4593 looks good for second review.

**Comment by Dusch4593** [comment]
Review 2 underway! 😄

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:58]: Can we remove this comment and write the following below this code-snippet?

The output will be:

```
Naomi is in engineering practicing her welding
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:65]:
```suggestion
In the example below, the `check_leap_year()` function returns a string which indicates if the passed parameter is a leap year or not.
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:68]: This seems like an unfinished codebyte snippet? Missing the body of `check_leap_year()`.

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:85]:
```suggestion
In the `calc_total()` function, the default value of `discount` is 10.
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:101]:
```suggestion
Unless specified otherwise, the arguments passed into a function are assigned to each parameter in the order in which they appear in the function signature. Thus, they are also known as "positional arguments".
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:111]: Same thing here. Can we move the comment out of the snippet and rewrite like this?

The output will be:

```
Jonas was wearing a yellow raincoat."
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:114]: What kind of error? Can we show in a comment?

```py
write_a_book(name="Jonas", "yellow", "raincoat") // Output: error message here...
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:140]: Can we change this snippet to not include political figures? Maybe change it to `north_american_capitals`? And let's also rewrite the output snippet like suggested previously:

```py
def north_american_capitals(**kwargs):
  ...

north_american_capitals(canada="Ottawa", us="Washington D.C.", mexico="Mexico City")
```

The output of the function call will be:

```
canada: Ottawa
us: Washington D.C.
mexico: Mexico City
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:192]: Can we show some example output of running this code?

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:195]:
```suggestion
Inside the function definition, in the `else` block, two recursive calls to `fibonacci()` (representing the two previous numbers) are added and returned. Once `n` is equal to `0` or `1`, the base case (the `if` block) runs instead. 
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:242]:
```suggestion
Lambda expressions are most commonly used to define single-use functions which are passed to higher-order functions.
```

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:260]: If the function can be written a single line, `lambda` could be useful. Otherwise, if the function is more complex and spans multiple lines, it is recommended to use the `def` keyword.

File: content/python/concepts/functions/functions.md
Comment (Dusch4593) [inline content/python/concepts/functions/functions.md:257]:
```suggestion
# Output:  2019
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@kaifee-haque I'm done with my review! 🏷️ Tag me when you're finished with resolving the comments/suggestions and we'll proceed! 😃 🏅

**Comment by kaifee-haque** [comment]
Thanks Brandon! All feedback incorporated.

**Review by Dusch4593 — APPROVED** [review]
Hi @kaifee-haque! Thanks for making those changes! Everything looks good to me! 👍 

I pushed a few additional changes that I wanted to outline here:

- For things like output and barebones syntax, let's use ``` instead of ```pseudo.
- I slightly changed the wording in the beginning of "## Function Parameters" to state that functions often need data from the outside to produce a desired result and that that is what parameters are intended to do.
- I changed `special_skill` back to `skill` since the other parameter examples are one word. I also changed `engineering` to `engineering school`. Lastly, I added a period in the `print()` statement to match what the output was saying. ("Naomi is in engineering school practicing her welding.")
- On line 70, I added the missing ` to `check_leap_year()`.

How does this look for merging, @sonnynomnom?

**Comment by kaifee-haque** [comment]
Thanks @Dusch4593! Though I just want to clarify that "Naomi in engineering" is a little nod to _The Expanse_ (engineering is a section within a spaceship) 😉

**Comment by yangc95** [comment]
@sonnynomnom did the third review on this entry 🍡  Wondering if we can use double quotes in place of italics, that have been taken out of the docs style guide, to emphasize keywords.

@kaifee-haque I made some small edits and changed up the metadata's definition.

**Review by yangc95 — APPROVED** [review]

**Comment by sonnynomnom** [comment]
Hi @kaifee-haque, your edited entry is live! https://www.codecademy.com/resources/docs/python/functions

I also made some more changes: https://github.com/Codecademy/docs/commit/4939bf30fbff430661ba8487eb49126efeb8a0ea

@SSwiniarski @Dusch4593 @yangc95 
- Make sure to make the output code block ` ` ` shell
- The error/exception code block ` ` ` error
- The pseudocode block ` ` ` pseudo
- And only two courses/Paths in ContentCatalog

**Pull request merged**
